palette/src/luv.rs

```diff
@@ -0,0 +1,857 @@
+use core::marker::PhantomData;
+use core::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign};
+
+#[cfg(feature = "random")]
+use rand::distributions::uniform::{SampleBorrow, SampleUniform, Uniform, UniformSampler};
+#[cfg(feature = "random")]
+use rand::distributions::{Distribution, Standard};
+#[cfg(feature = "random")]
+use rand::Rng;
+
+use crate::convert::FromColorUnclamped;
+use crate::encoding::pixel::RawPixel;
+use crate::white_point::{WhitePoint, D65};
+use crate::{
+    clamp, contrast_ratio, from_f64, Alpha, Clamp, Component, ComponentWise, FloatComponent,
+    GetHue, LuvHue, Mix, Pixel, RelativeContrast, Shade, Xyz,
+};
+
+/// CIE L\*u\*v\* (CIELUV) with an alpha component. See the [`Luva`
+/// implementation in `Alpha`](crate::Alpha#Luva).
+pub type Luva<Wp = D65, T = f32> = Alpha<Luv<Wp, T>, T>;
+
+/// The CIE L\*u\*v\* (CIELUV) color space.
+///
+/// CIE L\*u\*v\* is a device independent color space. It is a simple
+/// transformation of the CIE XYZ color space with the additional
+/// property of being more perceptually uniform. In contrast to
+/// CIELAB, CIELUV is also linear for a fixed lightness: additive
+/// mixtures of colors (at a fixed lightness) will fall on a line in
+/// CIELUV-space.
+///
+/// As a result, CIELUV is used more frequently for additive settings.
+#[derive(Debug, Pixel, FromColorUnclamped, WithAlpha)]
+#[cfg_attr(feature = "serializing", derive(Serialize, Deserialize))]
+#[palette(
+    palette_internal,
+    white_point = "Wp",
+    component = "T",
+    skip_derives(Xyz, Luv)
+)]
+#[repr(C)]
+pub struct Luv<Wp = D65, T = f32>
+where
+    T: FloatComponent,
+    Wp: WhitePoint,
+{
+    /// L\* is the lightness of the color. 0.0 gives absolute black and 100
+    /// give the brightest white.
+    pub l: T,
+
+    /// u* goes from -100 to 100.
+    pub u: T,
+
+    /// v* goes from -100 to 100.
```

Comment:
It's probably good to escape the `*` for good measure, to avoid accidental emphasis formatting.